packages/utils/src/structs/checkpoint.cairo

```diff
@@ -3,18 +3,21 @@
 
 use core::num::traits::Sqrt;
 use crate::math;
+use starknet::storage::{StoragePath, StorageAsPath, Vec, VecTrait, Mutable, MutableVecTrait};
+use starknet::storage::{StoragePointerReadAccess, StoragePointerWriteAccess};
 use starknet::storage_access::StorePacking;
-use super::storage_array::{StorageArray, StorageArrayTrait};
+
+const _2_POW_184: felt252 = 0x10000000000000000000000000000000000000000000000;
 
 /// `Trace` struct, for checkpointing values as they change at different points in
 /// time, and later looking up past values by block timestamp.
-#[derive(Copy, Drop, starknet::Store)]
+#[starknet::storage_node]
 pub struct Trace {
-    pub checkpoints: StorageArray<Checkpoint>
+    pub checkpoints: Vec<Checkpoint>
 }
 
 /// Generic checkpoint representation.
-#[derive(Copy, Drop, Serde)]
+#[derive(Copy, Drop, Serde, Debug, PartialEq)]
```

Comment:
Deriving from `Debug` and `PartialEq` made it easier to write some of the tests, but I'm not sure if it has any meaningful impact in gas costs since this is saved in storage(and didn't find anything online)
@ericnordelo @ericnordelo raising it in case you are aware of anything

Reply:
The compiler should be smart enough to optimize the generated implementations away as long as they are not used in the release target. With this said, I personally don't like if we start adding Debug and PartialEq to every struct just for testing purposes (unless we really need to). I vote we remove at least Debug since that's basically used just for printing to standard output, and PartialEq as well, since we don't really need it, and users can easily implement it if they do.

Reply:
I went ahead and removed it, it just made one of the tests slightly uglier. If the compiler gets rid of this, then I think we should consider deriving some of these traits, both for our tests and for users wanting to use them, but we can take that discussion offline.